File: 2. Add Two Numbers/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func addTwoNumbers(l1, l2 *ListNode) *ListNode {
	p := l1
	q := l2
	for {
		if p == nil {
			break
		}
		n := p.Val
		if q != nil {
			n += q.Val
		}
		p.Val = n % 10
		if n >= 10 {
			if p.Next == nil {
				p.Next = &ListNode{}
			}
			p.Next.Val++
		}
		if p.Next == nil && q != nil {
			p.Next = q.Next
			break
		}
		p = p.Next
		if q != nil {
			q = q.Next
		}
	}
	return l1
}

File: 2. Add Two Numbers/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
  def addTwoNumbers(self, l1, l2):
    """
    :type l1: ListNode
    :type l2: ListNode
    :rtype: ListNode
    """
    p = l1
    q = l2
    t = None
    c = 0

    while p and q:
      p.val = p.val + q.val + c
      c = p.val // 10
      p.val = p.val % 10
      t = p
      p = p.next
      q = q.next

    while p:
      p.val = p.val + c
      c = p.val // 10
      p.val = p.val % 10
      t = p
      p = p.next

    p = t
    p.next = q
    p = p.next

    while p:
      p.val = p.val + c
      c = p.val // 10
      p.val = p.val % 10
      t = p
      p = p.next

    if c == 1:
      p = t
      x = ListNode(1)
      p.next = x

    return l1